library/azure_rm_hdinsightcluster.py

```diff
@@ -0,0 +1,575 @@
+            domain_username:
+                description:
+                    - The I(domain) user account that will have admin privileges on the cluster.
+            domain_user_password:
+                description:
+                    - The I(domain) admin password.
+            cluster_users_group_dns:
+                description:
+                    - Optional. The Distinguished Names for cluster user groups
+                type: list
+            aadds_resource_id:
+                description:
+                    - "The resource ID of the user's Azure Active Directory I(domain) Service."
+            msi_resource_id:
+                description:
+                    - "User assigned identity that has permissions to read and create cluster-related artifacts in the user's AADDS."
+    compute_profile_roles:
+        description:
+            - The list of roles in the cluster.
+        type: list
+        suboptions:
+            name:
+                description:
+                    - The name of the role.
+            min_instance_count:
+                description:
+                    - The minimum instance count of the cluster.
+            target_instance_count:
+                description:
+                    - The instance count of the cluster.
+            vm_size:
+                description:
+                    - The size of the VM
+            linux_profile:
+                description:
+                    - The Linux OS profile.
+            virtual_network_profile:
+                description:
+                    - The virtual network profile.
+                suboptions:
+                    id:
+                        description:
+                            - The ID of the virtual network.
+                    subnet:
+                        description:
+                            - The name of the subnet.
+            data_disks_groups:
+                description:
+                    - The data disks groups for the role.
+                type: list
+                suboptions:
+                    disks_per_node:
+                        description:
+                            - The number of disks per node.
+            script_actions:
+                description:
+                    - The list of script actions on the role.
+                type: list
+                suboptions:
+                    name:
+                        description:
+                            - The name of the script action.
+                            - Required when C(state) is I(present).
+                    uri:
+                        description:
+                            - The URI to the script.
+                            - Required when C(state) is I(present).
+                    parameters:
+                        description:
+                            - The parameters for the script provided.
+                            - Required when C(state) is I(present).
+    storage_accounts:
```

Comment:
again, 3 types; resource id, resource group+ name, name.  Does storage account in another resource group supported?